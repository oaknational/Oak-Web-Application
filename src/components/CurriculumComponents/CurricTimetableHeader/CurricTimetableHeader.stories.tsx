import { Meta, StoryObj } from "@storybook/react";
import {
  OakBox,
  oakDefaultTheme,
  OakThemeProvider,
} from "@oaknational/oak-components";

import { CurricShowSteps } from "../CurricShowSteps";

import { CurricTimetableHeader as Component } from ".";

import illustrations from "@/image-data/generated/illustrations.json";

const meta: Meta<typeof Component> = {
  component: Component,
  tags: ["autodocs"],
  argTypes: {
    titleSlot: {
      type: "string",
      description: "String or ReactNode",
    },
    additionalSlot: {
      type: "string",
      description: "String or ReactNode",
    },
    illustrationSlug: {
      control: "select",
      options: Object.keys(illustrations),
    },
  },
  parameters: {
    controls: {
      include: ["titleSlot", "additionalSlot", "illustrationSlug"],
    },
  },
  decorators: [
    (Story) => (
      <OakThemeProvider theme={oakDefaultTheme}>
        <Story />
      </OakThemeProvider>
    ),
  ],
  render: (args) => <Component {...args} />,
};

export default meta;
type Story = StoryObj<typeof Component>;

export const Default: Story = {
  args: {
    titleSlot: "Year 1 maths",
    illustrationSlug: "magic-carpet",
    additionalSlot: "Additional content goes here...",
  },
};

export const WithSteps: Story = {
  args: {
    titleSlot: "Year 1 history",
    illustrationSlug: "magic-carpet",
    additionalSlot: (
      <OakBox $maxWidth={"all-spacing-20"}>
        <CurricShowSteps numberOfSteps={4} currentStepIndex={1} />
      </OakBox>
    ),
  },
};
