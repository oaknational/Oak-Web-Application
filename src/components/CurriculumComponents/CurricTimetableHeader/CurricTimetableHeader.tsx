import {
  OakBox,
  OakFlex,
  OakHeading,
  OakImage,
} from "@oaknational/oak-components";
import React from "react";

import Illustration from "@/components/SharedComponents/Illustration";
import { IllustrationSlug } from "@/image-data";

const logoSrc = `https://cdn.sanity.io/images/cuvjke51/production/e9cc7ae43ca35576d95d0c3ef22f76e0aee14eac-1920x880.svg`;

type CurricTimetableHeaderProps = {
  titleSlot: React.ReactNode;
  additionalSlot?: React.ReactNode;
  illustrationSlug: IllustrationSlug;
};
export function CurricTimetableHeader({
  titleSlot,
  additionalSlot,
  illustrationSlug,
}: CurricTimetableHeaderProps) {
  return (
    <OakBox
      $background={"bg-decorative1-main"}
      $borderRadius={"border-radius-xl"}
      $pa={"inner-padding-xl5"}
      $pt={"inner-padding-xl4"}
      $pb={"inner-padding-xl3"}
    >
      <OakFlex $flexDirection={"row"}>
        <OakFlex $flexDirection={"column"} $flexGrow={1}>
          <OakFlex $pb={"inner-padding-s"} $alignItems={"start"}>
            <OakImage
              src={logoSrc}
              style={{ height: 52, width: 114 }}
              alt="Oak National Academy"
            />
          </OakFlex>
          <OakFlex>
            <OakFlex
              $gap={"all-spacing-9"}
              $flexDirection={"column"}
              $flexGrow={1}
            >
              <OakHeading
                tag="h1"
                $font={"heading-2"}
                data-testid="timetable-header-title"
              >
                {titleSlot}
              </OakHeading>
              <OakBox data-testid="timetable-header-additional">
                {additionalSlot}
              </OakBox>
            </OakFlex>
          </OakFlex>
        </OakFlex>
        <OakFlex
          $display={["none", "none", "flex"]}
          $flexGrow={1}
          $flexShrink={1}
          $position={"relative"}
        >
          <Illustration
            slug={illustrationSlug}
            $objectFit="contain"
            $objectPosition={"bottom right"}
            fill
            $height={"100%"}
          />
        </OakFlex>
      </OakFlex>
    </OakBox>
  );
}
