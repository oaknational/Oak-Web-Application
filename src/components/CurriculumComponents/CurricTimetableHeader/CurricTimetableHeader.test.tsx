import { CurricTimetableHeader } from "./CurricTimetableHeader";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("CurricTimetableHeader", () => {
  test("basic usage", () => {
    const { baseElement, getByTestId } = renderWithTheme(
      <CurricTimetableHeader
        titleSlot={"TITLE"}
        additionalSlot={"ADDITIONAL"}
        illustrationSlug={"magic-carpet"}
      />,
    );
    expect(baseElement).toMatchSnapshot();
    expect(getByTestId("timetable-header-title")).toHaveTextContent("TITLE");
    expect(getByTestId("timetable-header-additional")).toHaveTextContent(
      "ADDITIONAL",
    );
  });
});
