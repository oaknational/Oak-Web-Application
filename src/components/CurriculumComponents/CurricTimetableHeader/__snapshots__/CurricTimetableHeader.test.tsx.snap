// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurricTimetableHeader basic usage 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR iYZQrv"
    >
      <div
        class="sc-fqkvVR sc-kAyceB fgBWrW dnkkZD"
      >
        <div
          class="sc-fqkvVR sc-kAyceB fgBWrW jOTTNg"
        >
          <div
            class="sc-fqkvVR sc-kAyceB cdQSNK cnUbCk"
          >
            <div
              class="sc-fqkvVR jqqmXN"
              style="height: 52px; width: 114px;"
            >
              <img
                alt="Oak National Academy"
                class="sc-dcJsrY jRtmJB"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                src="https://cdn.sanity.io/images/cuvjke51/production/e9cc7ae43ca35576d95d0c3ef22f76e0aee14eac-1920x880.svg"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
          <div
            class="sc-fqkvVR sc-kAyceB fgBWrW joHkzz"
          >
            <div
              class="sc-fqkvVR sc-kAyceB fgBWrW iJXlvX"
            >
              <h1
                class="sc-kpDqfm jGrEMo"
                data-testid="timetable-header-title"
              >
                TITLE
              </h1>
              <div
                class="sc-fqkvVR fgBWrW"
                data-testid="timetable-header-additional"
              >
                ADDITIONAL
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fqkvVR sc-kAyceB bygFWn dnFLNa"
        >
          <img
            alt=""
            class="sc-6170517-0 gnvkSN"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=3840&fm=webp&q=80&fit=clip&auto=format"
            srcset="https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=640&fm=webp&q=80&fit=clip&auto=format 640w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=750&fm=webp&q=80&fit=clip&auto=format 750w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=828&fm=webp&q=80&fit=clip&auto=format 828w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=1080&fm=webp&q=80&fit=clip&auto=format 1080w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=1200&fm=webp&q=80&fit=clip&auto=format 1200w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=1920&fm=webp&q=80&fit=clip&auto=format 1920w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=2048&fm=webp&q=80&fit=clip&auto=format 2048w, https://NEXT_PUBLIC_SANITY_ASSET_CDN_HOST/images/NEXT_PUBLIC_SANITY_PROJECT_ID/NEXT_PUBLIC_SANITY_DATASET/897fadaf636cdb351e6213115fde07c2d7eb771f-1588x872.svg?w=3840&fm=webp&q=80&fit=clip&auto=format 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
